File: VB/CR_CodeDecorationBlueBox/CR_CodeDecorationBlueBox/BlueBoxAdornment.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports DevExpress.DXCore.Adornments
Imports DevExpress.DXCore.Platform.Drawing

Public Class BlueBoxAdornment
    Inherits VisualObjectAdornment

    Public Sub New(ByVal feature As String, ByVal frame As IElementFrame)
        MyBase.New(feature, frame)
    End Sub

    Public Overrides Sub Render(ByVal context As IDrawingSurface, ByVal geometry As ElementFrameGeometry)
        context.DrawRectangle(Nothing, Colors.Blue, geometry.Bounds)
    End Sub
End Class

File: CS/CR_CodeDecorationBlueBox/CR_CodeDecorationBlueBox/BlueBoxAdornment.cs
using System;
using System.Collections.Generic;
using DevExpress.DXCore.Adornments;
using DevExpress.DXCore.Platform.Drawing;
using DevExpress.CodeRush.Core;
using DevExpress.CodeRush.StructuralParser;

namespace CR_CodeDecorationBlueBox
{
    class BlueBoxAdornment : VisualObjectAdornment
    {
        public BlueBoxAdornment(string feature, IElementFrame frame)
            : base(feature, frame)
        {
        }
        public override void Render(IDrawingSurface context, ElementFrameGeometry geometry)
        {                
            // TODO: Add adornment painting logic
            context.DrawRectangle(null, Colors.Blue, geometry.Bounds);
        }
    }
}
